### PR TITLE
tests: Use LvlInitStruct<> consistently in descriptor tests

### PR DESCRIPTION
**Pull request by jeremyg-lunarg**
Cleanup change to always use `auto foo = LvlInitStruct<VkFoo>()` in these tests. Also adds the ability to intialize other fields in the LvlInitStruct<> call and a size32() helper to make it more concise to set size fields container size() methods.

Note: there's a follow on change to use vk_testing:: objects in these tests, but it is too much for one PR.

### CODE REVIEW

File: layers/generated/vk_typemap_helper.h [hunk 1]
```diff
@@ -7029,19 +7029,16 @@ template <typename T> T *LvlFindModInChain(void *next) {
...
 
-// Init the header of an sType struct with pNext
-template <typename T> T LvlInitStruct(void *p_next) {
-    T out = {};
-    out.sType = LvlTypeMap<T>::kSType;
-    out.pNext = p_next;
+// Init the header of an sType struct with pNext and optional fields
+template <typename T, typename... StructFields>
+T LvlInitStruct(void *p_next, StructFields... fields) {
+    T out = {LvlTypeMap<T>::kSType, p_next, fields...};
```
Comment (ncesario-lunarg) [inline layers/generated/vk_typemap_helper.h:7035]: I was just wishing I had this yesterday. This is awesome!

File: tests/layer_validation_tests.h [hunk 2]
```diff
@@ -184,6 +184,16 @@ size_t size(ElementT (&)[array_size]) {
     return array_size;
 }
 
+template <class ElementT, size_t array_size>
+uint32_t size32(ElementT (&)[array_size]) {
```
Comment (ncesario-lunarg) [inline tests/layer_validation_tests.h:188]: 🎉

**Review by ncesario-lunarg — APPROVED** [review]
+100 for this change

File: tests/vklayertests_descriptor_renderpass_framebuffer.cpp [hunk 3]
```diff
@@ -174,17 +174,7 @@ TEST_F(VkLayerTest, DrawWithPipelineIncompatibleWithSubpass) {
...
-                                  rp,
-                                  fb,
-                                  {{
-                                       0,
-                                       0,
-                                   },
-                                   {32, 32}},
-                                  0,
-                                  nullptr};
+    auto rpbi = LvlInitStruct<VkRenderPassBeginInfo>(nullptr, rp, fb, VkRect2D{{0, 0}, {32u, 32u}}, 0u, nullptr);
```
Comment (Tony-LunarG) [inline tests/vklayertests_descriptor_renderpass_framebuffer.cpp:177]: Cool!

**Review by Tony-LunarG — APPROVED** [review]
LGTM!

File: layers/generated/vk_typemap_helper.h (hunk 1 above)
Comment (spencer-lunarg) [reply layers/generated/vk_typemap_helper.h:7035]: :pray:

**Pull request merged**